odl/discr/grid.py

```diff
@@ -47,23 +47,28 @@ def sparse_meshgrid(*x, **kwargs):
     order = kwargs.pop('order', 'C')
     mg = []
     for ax, xi in enumerate(x):
+        xi = np.asarray(xi)
```

Comment:
I've never benchmarked it. Here the situation is that either one gets a bunch of `ndarrays` already, in which case it's a no-op (I think), otherwise there's something array-like and we need to do something anyway. I added this line just to be able to use array-like objects instead of just `ndarrays`.